web/src/ShareSnapshotModal.tsx

```diff
@@ -12,62 +12,108 @@ type props = {
+          </p>
+          {link}
+          <p className="ShareSnapshotModal-user">
+            Sharing as <strong>{user}</strong>
+          </p>
+        </div>
         <div className="ShareSnapshotModal-pane">{this.tiltCloudCopy()}</div>
       </Modal>
     )
   }
 
   renderShareLink() {
     const hasLink = this.props.snapshotUrl !== ""
-    if (!hasLink) {
-      return (
-        <button onClick={this.props.handleSendSnapshot}>Share Snapshot</button>
-      )
-    }
     return (
-      <a href={this.props.snapshotUrl} target="_blank">
-        Open link in new tab
-      </a>
+      <section className="ShareSnapshotModal-shareLink">
+        <div className="ShareSnapshotModal-shareLink-urlBox">
+          {hasLink ? (
+            <p className="ShareSnapshotModal-shareLink-url">
+              {this.props.snapshotUrl}
+            </p>
+          ) : (
+            <p className="ShareSnapshotModal-shareLink-placeholder">
+              No Link Generated Yet
+            </p>
+          )}
+        </div>
+        {!hasLink && (
+          <button
+            className="ShareSnapshotModal-button"
+            onClick={this.props.handleSendSnapshot}
+          >
+            Get Link
+          </button>
+        )}
+        {hasLink && (
+          <a
+            className="ShareSnapshotModal-button"
+            href={this.props.snapshotUrl}
+            target="_blank"
+          >
+            Open
+          </a>
+        )}
+      </section>
     )
   }
 
   tiltCloudCopy() {
     if (!this.props.tiltCloudUsername) {
       return (
-        <div>
-          <div>
-            Click{" "}
-            <form
-              action={this.props.tiltCloudSchemeHost + "/register_token"}
-              target="_blank"
-              method="POST"
-              onSubmit={ShareSnapshotModal.notifyTiltOfRegistration}
-            >
-              <input type="submit" value="here" />
-              <input
-                name="token"
-                type="hidden"
-                value={cookies.get("Tilt-Token")}
-              />
-            </form>
-            to associate this copy of Tilt with your TiltCloud account.
-          </div>
-          <div>(You'll just need to link your GitHub Account)</div>
-        </div>
+        <section>
+          <p>
+            Register on TiltCloud to share under your name, view, and delete
+            snapshots.
+          </p>
+          <form
+            className="ShareSnapshotModal-tiltCloudButtonForm"
+            action={this.props.tiltCloudSchemeHost + "/register_token"}
+            target="_blank"
+            method="POST"
+            onSubmit={ShareSnapshotModal.notifyTiltOfRegistration}
+          >
+            <p>You'll need a GitHub account</p>
+            <span className="ShareSnapshotModal-tiltCloudButtonForm-spacer" />
+            <input
+              className="ShareSnapshotModal-tiltCloudButton"
+              type="submit"
+              value="Sign Up"
```

Comment:
this copy isn't quiiiite right. needs to be like "sign up / link account" or something (cuz they may already be signed up for tilt cloud, and it's just associated with this Tilt instance yet)

Reply:
I agree, the "Sign Up" is misleading. I will consider a better way to express that you're linking your Tilt instance.